### PR TITLE
adding ok state metric filter

### PR DESCRIPTION
**Pull request by jcdietrich**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -152,7 +152,7 @@ class Plugin {
 		var output = {};
 		if (alarm.pattern) {
 			//add custom log metric
-			const logMetricCFRef = this.naming.getLogMetricCFRef(normalizedFunctionName,alarm.name);
+			const logMetricCFRef = this.naming.getLogMetricCFRef(normalizedFunctionName,alarm.name)+'ALERT';
```
Comment (AntonBazhal) [inline src/index.js]: Use [template literal](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals) here

File: src/index.js [hunk 2]
```diff
@@ -167,8 +167,24 @@ class Plugin {
 					}]
 				}
 			};
+			//add custom log metric for ok state
+			const logMetricCFRefOK = this.naming.getLogMetricCFRef(normalizedFunctionName,alarm.name)+'OK';
```
Comment (AntonBazhal) [inline src/index.js]: template literal

File: src/index.js [hunk 3]
```diff
@@ -167,8 +167,24 @@ class Plugin {
...
+			output[logMetricCFRefOK] = {
+				Type: "AWS::Logs::MetricFilter",
+				DependsOn: CFLogName,
+				Properties: {
+					FilterPattern: "",
+					LogGroupName: this.serverless.getProvider('aws').naming.getLogGroupName(this.serverless.service.getFunction(functionName).name),
+					MetricTransformations: [{
+						MetricValue: 0,
+						MetricNamespace: `${this.serverless.service.service}_${alarm.namespace}`,
+						MetricName: alarm.metric + normalizedFunctionName
```
Comment (AntonBazhal) [inline src/index.js]: template literal and extract this to variable (the same for alarm)

File: src/index.js [hunk 4]
```diff
@@ -167,8 +167,24 @@ class Plugin {
...
+			const logMetricCFRefOK = this.naming.getLogMetricCFRef(normalizedFunctionName,alarm.name)+'OK';
+			output[logMetricCFRefOK] = {
+				Type: "AWS::Logs::MetricFilter",
+				DependsOn: CFLogName,
+				Properties: {
+					FilterPattern: "",
+					LogGroupName: this.serverless.getProvider('aws').naming.getLogGroupName(this.serverless.service.getFunction(functionName).name),
+					MetricTransformations: [{
+						MetricValue: 0,
+						MetricNamespace: `${this.serverless.service.service}_${alarm.namespace}`,
```
Comment (AntonBazhal) [inline src/index.js]: extract this to variable (the same for alarm)

File: src/index.js [hunk 5]
```diff
@@ -167,8 +167,24 @@ class Plugin {
...
 				}
 			};
+			//add custom log metric for ok state
+			const logMetricCFRefOK = this.naming.getLogMetricCFRef(normalizedFunctionName,alarm.name)+'OK';
+			output[logMetricCFRefOK] = {
+				Type: "AWS::Logs::MetricFilter",
+				DependsOn: CFLogName,
+				Properties: {
+					FilterPattern: "",
+					LogGroupName: this.serverless.getProvider('aws').naming.getLogGroupName(this.serverless.service.getFunction(functionName).name),
```
Comment (AntonBazhal) [inline src/index.js]: extract this to variable (the same for alarm)

File: src/index.js [hunk 6]
```diff
@@ -151,33 +151,38 @@ class Plugin {
...
 				DependsOn: CFLogName,
 				Properties: {
 					FilterPattern: alarm.pattern,
-					LogGroupName: this.serverless.getProvider('aws').naming.getLogGroupName(this.serverless.service.getFunction(functionName).name),
+					LogGroupName: logGroupName,
 					MetricTransformations: [{
 						MetricValue: 1,
-						MetricNamespace: `${this.serverless.service.service}_${alarm.namespace}`,
+						MetricNamespace: metricNamespace,
 						MetricName: alarm.metric + normalizedFunctionName
```
Comment (AntonBazhal) [inline src/index.js]: I think, this is the same for both metrics.

Comment (jcdietrich) [inline src/index.js]: it should be. I need them both to post the the same metric namespace and metric name. one posting 0 and the other 1.

Comment (AntonBazhal) [inline src/index.js]: I mean, why not to extract variable then?

**Pull request merged**